components/supervisor/pkg/supervisor/services.go

```diff
@@ -38,11 +38,42 @@ type RegisterableRESTService interface {
 	RegisterREST(mux *runtime.ServeMux, grpcEndpoint string) error
 }
 
+type ideReadyService struct {
+	ready bool
+	cond  *sync.Cond
+}
+
+func (service *ideReadyService) wait(ctx context.Context) {
```

Comment:
nit: even though the `ideReadyService`/State is entirely package-internal, personally I'd make `Wait`, `Get` and `Set` uppercase to indicate that they are meant for consumption.
Also, a quick one-line function comment would not hurt :)